components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -343,6 +343,15 @@ public String streamingPredict(int tenantId, String userName, long modelId, Stri
         }
 
         MLModel builtModel = retrieveModel(modelId);
+
+        // Validate number of features in predict dataset
+        if (builtModel.getFeatures().size() != data.get(0).length) {
+            String msg = String.format("Failed to predict from model. ML model with model id: %s has %s features " +
+                    "but predict dataset has %s features. Incompatible number of features.", modelId,
+                    builtModel.getFeatures().size(), data.get(0).length);
```

Comment:
Shall we change the message to something like;
Prediction failed from model [id] x since the number of features of model [4] doesn't match the number of features in the input data [3]. 

Wdyt?